test/testjsonld-cpp/CMakeLists.txt

```diff
@@ -1,66 +1,123 @@
-
-target_compile_features(UnitTests_jsonld-cpp PRIVATE cxx_std_11)
-target_compile_options(UnitTests_jsonld-cpp PRIVATE ${DCD_CXX_FLAGS})
-set_target_properties(UnitTests_jsonld-cpp PROPERTIES CXX_EXTENSIONS OFF)
-
-target_include_directories(UnitTests_jsonld-cpp
-        PUBLIC
-        ${PROJECT_SOURCE_DIR}/libjsonld-cpp)
-
-target_link_libraries(UnitTests_jsonld-cpp jsonld-cpp Boost::filesystem gtest gmock rapidcheck_gtest)
-
+#
+# Build common testfiled into lib so they only need to be compiled once
+#
+add_library(
+    libjsonld-cpp_test STATIC
+    test_FileLoader.cpp
+    main.cpp
+    testHelpers.cpp
+    testHelpers.h
+    ManifestLoader.cpp
+    ManifestLoader.h
+    UrdnaManifestLoader.cpp
+    UrdnaManifestLoader.h
+    TestCase.cpp
+    TestCase.h
+    UriBaseRewriter.cpp
+    UriBaseRewriter.h
+    TestCaseOptions.cpp
+    TestCaseOptions.h
+)
+
+target_include_directories(
+    libjsonld-cpp_test
+    PUBLIC
+    ${PROJECT_SOURCE_DIR}/libjsonld-cpp
+)
+
+target_link_libraries(
+    libjsonld-cpp_test
+    jsonld-cpp
+    Boost::filesystem
+    gtest
+    gmock
+    rapidcheck_gtest
+)
+
+#
+# jsonld-cpp
+#
+add_executable(
+    UnitTests_jsonld-cpp
+    test_JsonLdApi.cpp
+    test_JsonLdUtils.cpp
+    test_NodeComparisons.cpp
+    test_ObjectComparisons.cpp
+    test_UniqueNamer.cpp
+    test_DoubleFormatter.cpp
+    test_Permutator.cpp
+    test_NormalizeUtils.cpp
+    test_Sha1.cpp
+    test_Regex.cpp
+    test_RDFDataset.cpp
+    test_RDFDatasetUtils.cpp
+    test_Uri.cpp
+    test_UriParser.cpp
+    test_Sha256.cpp
+)
+
+target_compile_features(
+    UnitTests_jsonld-cpp
+    PRIVATE
+    cxx_std_11
+)
+target_compile_options(
+    UnitTests_jsonld-cpp
+    PRIVATE
+    ${DCD_CXX_FLAGS}
+)
+set_target_properties(
+    UnitTests_jsonld-cpp
+    PROPERTIES
+    CXX_EXTENSIONS OFF
+)
+target_include_directories(
+    UnitTests_jsonld-cpp
+    PUBLIC
+    ${PROJECT_SOURCE_DIR}/libjsonld-cpp
+)
+target_link_libraries(
+    UnitTests_jsonld-cpp
+    libjsonld-cpp_test
+)
 add_test(NAME UnitTests_jsonld-cpp
         COMMAND UnitTests_jsonld-cpp)
 
-####
-
-add_executable(UnitTests_JsonLdProcessor_expand_jsonld-cpp main.cpp test_JsonLdProcessor_expand.cpp testHelpers.cpp testHelpers.h ManifestLoader.cpp ManifestLoader.h UrdnaManifestLoader.cpp UrdnaManifestLoader.h TestCase.cpp TestCase.h UriBaseRewriter.cpp UriBaseRewriter.h TestCaseOptions.cpp TestCaseOptions.h)
-
-target_compile_features(UnitTests_JsonLdProcessor_expand_jsonld-cpp PRIVATE cxx_std_11)
-target_compile_options(UnitTests_JsonLdProcessor_expand_jsonld-cpp PRIVATE ${DCD_CXX_FLAGS})
-set_target_properties(UnitTests_JsonLdProcessor_expand_jsonld-cpp PROPERTIES CXX_EXTENSIONS OFF)
-
-target_include_directories(UnitTests_JsonLdProcessor_expand_jsonld-cpp
+#
+# API
+#
+set(ALGORITHMS "expand;toRDF;normalize-urdna2015")
+
+foreach(ALGO ${ALGORITHMS})
```

Comment:
no way - this is awesome

Reply:
It's highly dependent on the naming convention but works quite well in this case